### PR TITLE
test(tagstore): Run tests with SnubaCompatibilityTagStorage backend

### PR DESCRIPTION
**Pull request by lynnagara**
This is the first step to removing the LegacyTagStorage tagstore backend. We do not use this backend anymore except for running tests, and it will not work properly when we deprecate the Postgres events table.

### CODE REVIEW

**Comment by fpacifici** [comment]
What's the goal with this one ?

**Comment by lynnagara** [comment]
The legacy and v2 backends are not used anywhere apart from the test suite anymore. The new style event ID is incompatible with these backends without a bit of work. Opening this PR to see how easy it will be to switch over and completely remove those legacy backends.... There are quite a few failing tests though :(

**Review by fpacifici — CHANGES_REQUESTED** [review]
File "/home/travis/build/getsentry/sentry/src/sentry/tagstore/base.py", line 166, in get_or_create_tag_value
    raise NotImplementedError

Several test failure seem pertinent

Back to you to address those cases

File: tests/sentry/api/endpoints/test_project_tagkey_values.py [hunk 1]
```diff
@@ -33,11 +45,21 @@ def test_simple(self):
...
     def test_query(self):
-        project = self.create_project()
-        tagkey = tagstore.create_tag_key(project_id=project.id, environment_id=None, key="foo")
-        tagstore.create_tag_value(
-            project_id=project.id, environment_id=None, key="foo", value="bar"
-        )
+        if settings.SENTRY_TAGSTORE in [
+            "sentry.tagstore.snuba.SnubaCompatibilityTagStorage",
+            "sentry.tagstore.snuba.SnubaTagStorage",
+        ]:
```
Comment (markstory) [inline tests/sentry/api/endpoints/test_project_tagkey_values.py]: Should we start warning folks when they are using a tagstore that isn't one of these?

**Comment by lynnagara** [comment]
@fpacifici Well this was rather painful, but I believe I've addressed all of the failing tests now

File: tests/sentry/api/endpoints/test_project_tagkey_values.py (hunk 1 above)
Comment (lynnagara) [reply tests/sentry/api/endpoints/test_project_tagkey_values.py]: What kind of warning are you thinking of? I was planning to just completely delete Legacy immediately after this gets merged (v2 is already gone)

File: tests/sentry/api/endpoints/test_group_tags.py [hunk 2]
```diff
@@ -1,102 +0,0 @@
-from __future__ import absolute_import
-
-from sentry import tagstore
-from sentry.testutils import APITestCase
-
-
-class GroupTagsTest(APITestCase):
```
Comment (fpacifici) [inline tests/sentry/api/endpoints/test_group_tags.py]: Why was this removed instead of replacing tag creation with event store  to produce the same result ?

File: tests/sentry/api/endpoints/test_organization_user_issues.py [hunk 3]
```diff
@@ -86,5 +90,5 @@ def test_has_access(self):
         # now result should include results from team2/project2
         assert response.status_code == 200
         assert len(response.data) == 2
-        assert response.data[0]["id"] == six.text_type(self.group2.id)
+        assert response.data[0]["id"] == six.text_type(self.group3.id)
```
Comment (fpacifici) [inline tests/sentry/api/endpoints/test_organization_user_issues.py]: why did this work before ?

File: tests/sentry/deletions/test_tagkey.py [hunk 4]
```diff
@@ -1,97 +0,0 @@
-from __future__ import absolute_import
-
-from sentry import tagstore
-from sentry.tagstore.models import EventTag
-from sentry.models import ScheduledDeletion
-from sentry.tasks.deletion import run_deletion
-from sentry.testutils import TestCase
-
-
-class DeleteTagKeyTest(TestCase):
```
Comment (fpacifici) [inline tests/sentry/deletions/test_tagkey.py]: Why was this deleted instead of creating events?

File: tests/sentry/manager/tests.py [hunk 5]
```diff
@@ -14,37 +15,48 @@ def test_valid_only_message(self):
...
+    )
     def test_add_tags(self):
-        event = Group.objects.from_kwargs(1, message="rrr")
-        group = event.group
-        environment = self.create_environment()
-
-        with self.tasks():
-            Group.objects.add_tags(
-                group, environment, tags=[("foo", "bar"), ("foo", "baz"), ("biz", "boz")]
+        if settings.SENTRY_TAGSTORE in [
```
Comment (fpacifici) [inline tests/sentry/manager/tests.py]: What's the goal of this test if we are getting rid of the legacy tagstore ?

**Review by fpacifici — COMMENTED** [review]
A couple of clarifications, otherwise it is ok.
Did you by chance check how long it takes now to run the full CI test suite compared to before ?
We should ensure it does not make the CI time too long

**Comment by lynnagara** [comment]
I haven't noticed any significant differences in the time to run tests. The latest run on this branch is actually a little bit faster than master.

File: tests/sentry/api/endpoints/test_organization_user_issues_search.py [hunk 6]
```diff
@@ -2,53 +2,48 @@
...
-from sentry import tagstore
-from sentry.models import EventUser, OrganizationMemberTeam
-from sentry.testutils import APITestCase
+from sentry.models import OrganizationMemberTeam
+from sentry.testutils import APITestCase, SnubaTestCase
+from sentry.testutils.helpers.datetime import iso_format, before_now
 
 
-class OrganizationUserIssuesSearchTest(APITestCase):
+class OrganizationUserIssuesSearchTest(APITestCase, SnubaTestCase):
```
Comment (markstory) [inline tests/sentry/api/endpoints/test_organization_user_issues_search.py]: Should this move to the snuba directory to live with the other snuba endpoint tests?

Comment (lynnagara) [reply tests/sentry/api/endpoints/test_organization_user_issues_search.py]: All the test suites depend on and run Snuba now, but the Snuba tests are still scattered in both the Snuba and Backend groups. It's quite confusing since the tests for a single module are often split between two different test suites.

Now that SnubaTagStorage is the default tagstore, I wonder if it's actually worth moving tests out of the Snuba directory instead where no longer needed.

File: tests/sentry/manager/tests.py (hunk 5 above)
Comment (lynnagara) [reply tests/sentry/manager/tests.py]: deleted

File: tests/sentry/deletions/test_tagkey.py (hunk 4 above)
Comment (lynnagara) [reply tests/sentry/deletions/test_tagkey.py]: Sorry what do you mean? Since the TagKey model is part of the legacy tagstore and will be completely removed soon, I don't think the test for tagkey deletions is very relevant anymore.

File: tests/sentry/api/endpoints/test_organization_user_issues.py (hunk 3 above)
Comment (lynnagara) [reply tests/sentry/api/endpoints/test_organization_user_issues.py]: previous test only actually created 2 groups but more tags, test is more realistic now

File: tests/sentry/api/endpoints/test_group_details.py [hunk 7]
```diff
@@ -64,30 +63,6 @@ def test_with_qualified_short_id(self):
...
 
-    def test_with_first_release(self):
-        self.login_as(user=self.user)
-
-        group = self.create_group()
-        release = Release.objects.create(
-            organization_id=group.project.organization_id, version="1.0"
-        )
-        release.add_project(group.project)
-        tagstore.create_group_tag_value(
```
[the remark this answers is not in the document]
Comment (lynnagara) [reply tests/sentry/api/endpoints/test_group_details.py]: I moved this to the Snuba group previously, will move back

File: tests/sentry/api/endpoints/test_group_tags.py (hunk 2 above)
Comment (lynnagara) [reply tests/sentry/api/endpoints/test_group_tags.py]: Sorry, I did not delete it but renamed the file earlier. Reverted that change now.

File: tests/sentry/api/endpoints/test_group_details.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (fpacifici) [reply tests/sentry/api/endpoints/test_group_details.py]: ok

File: tests/sentry/api/endpoints/test_group_tags.py (hunk 2 above)
Comment (fpacifici) [reply tests/sentry/api/endpoints/test_group_tags.py]: ok

File: tests/sentry/deletions/test_tagkey.py (hunk 4 above)
Comment (fpacifici) [reply tests/sentry/deletions/test_tagkey.py]: ok, understood

**Review by fpacifici — APPROVED** [review]
